### PR TITLE
Migration to Kotlin

### PR DESCRIPTION
**Pull request by Arnold2381**
## Purpose / Description
Migration to kotlin.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/LocaleSelectionDialog.kt [hunk 1]
```diff
@@ -13,260 +13,193 @@
...
-                    Collection<? extends Locale> values = (Collection<? extends Locale>) results.values;
-                    visibleLocales.addAll(values);
-                    notifyDataSetChanged();
-                }
-            };
+    companion object {
+        /**
+         * @param handler Marker interface to enforce the convention the caller implementing LocaleSelectionDialogHandler
+         */
+        fun newInstance(handler: LocaleSelectionDialogHandler): LocaleSelectionDialog {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/LocaleSelectionDialog.kt:197]: Make this `@JvmStatic`, and it should be callable, there shouldn't be a need to refactor Java code in a Kotlin conversion

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
One issue, rest looks good

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
Exciting! Material Dialogs conversion begins :partying_face:

**Pull request merged**